packages/hardhat/contracts/periphery/ControllerHelper.sol

```diff
@@ -762,14 +762,11 @@ contract ControllerHelper is UniswapControllerHelper, EulerControllerHelper, IER
             ControllerHelperDataType.CALLBACK_SOURCE.SWAP_EXACTIN_WPOWERPERP_ETH
         ) {
             IWPowerPerp(wPowerPerp).transfer(_pool, _amountToPay);
-
-            if (address(this).balance > 0) IWETH9(weth).deposit{value: address(this).balance}();
```

Comment:
So there is no scenario where at this point some ETH still not wrapped?

Reply:
The only place I see ETH potentially being in that callback is in _closeShortWithAmountsFromLp().  However, I don't think it matters.  1) This callback is selling oSQTH for ETH (so it doesnt need to have any weth ready to pay for anything, just oSQTH. Additionally, after the callback in _closeShortWithAmountsFromLp(), we call wrapInternal() (and in all cases, sendBack() is called at the end of the transaction.

```
            } else {
                // if LP have more wPowerPerp amount that amount to burn in vault, sell remaining amount for WETH
                ControllerHelperUtil.withdrawFromVault(
                    controller,
                    weth,
                    _vaultId,
                    _wPowerPerpAmountToBurn,
                    _collateralToWithdraw
                );

                uint256 wPowerPerpExcess = _wPowerPerpAmount.sub(_wPowerPerpAmountToBurn);
                if (wPowerPerpExcess > 0) {
                    _exactInFlashSwap(
                        wPowerPerp,
                        weth,
                        _poolFee,
                        wPowerPerpExcess,
                        _limitPriceEthPerPowerPerp.mul(wPowerPerpExcess).div(1e18),
                        uint8(ControllerHelperDataType.CALLBACK_SOURCE.SWAP_EXACTIN_WPOWERPERP_ETH),
                        ""
                    );
                }
            }
        }

        // wrap ETH to WETH
        wrapInternal(address(this).balance);
```